src/Microsoft.ML.TensorFlow/Microsoft.ML.TensorFlow.csproj

```diff
@@ -0,0 +1,31 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <IncludeInPackage>Microsoft.ML.TensorFlow</IncludeInPackage>
+    <DefineConstants>CORECLR</DefineConstants>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.ML.Core\Microsoft.ML.Core.csproj" />
+    <ProjectReference Include="..\Microsoft.ML.Data\Microsoft.ML.Data.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Update="TensorFlow\TensorGeneric.cs">
+      <DesignTime>True</DesignTime>
+      <AutoGen>True</AutoGen>
+      <DependentUpon>TensorGeneric.tt</DependentUpon>
+    </Compile>
+    <None Update="TensorFlow\TensorGeneric.tt">
+      <Generator>TextTemplatingFileGenerator</Generator>
+      <LastGenOutput>TensorGeneric.cs</LastGenOutput>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{508349b6-6b84-4df5-91f0-309beebad82d}" />
```

Comment:
(minor) What does this do?  Is it necessary?

Reply:
That enables the T4 service.  Yes it is necessary.  You'll see the same in corefx: https://github.com/dotnet/corefx/search?q=508349b6-6b84-4df5-91f0-309beebad82d&unscoped_q=508349b6-6b84-4df5-91f0-309beebad82d

Reply:
question - what is the T4 service ?

---
In reply to: [214093573](https://github.com/dotnet/machinelearning/pull/704#discussion_r214093573) [](ancestors = 214093573)